exporter/trace/trace.go

```diff
@@ -44,23 +53,72 @@ func newTraceExporter(o *options) (*traceExporter, error) {
 		client:    client,
 		o:         o,
 	}
+	b := bundler.NewBundler((*contextAndSpans)(nil), func(bundle interface{}) {
+		ctxSpans := bundle.([]*contextAndSpans)
+		ctxToSpansMap := make(map[context.Context][]*tracepb.Span)
+		// upload spans with same context in batch
+		for _, cs := range ctxSpans {
+			ctxToSpansMap[cs.ctx] = append(ctxToSpansMap[cs.ctx], cs.spans...)
+		}
+		for ctx, spans := range ctxToSpansMap {
+			e.uploadFn(ctx, spans)
+		}
+	})
+	if o.BundleDelayThreshold > 0 {
+		b.DelayThreshold = o.BundleDelayThreshold
+	} else {
+		b.DelayThreshold = 2 * time.Second
+	}
+	if o.BundleCountThreshold > 0 {
+		b.BundleCountThreshold = o.BundleCountThreshold
+	} else {
+		b.BundleCountThreshold = 50
+	}
+	if o.NumberOfWorkers > 0 {
+		b.HandlerLimit = o.NumberOfWorkers
+	}
+	// The measured "bytes" are not really bytes, see exportReceiver.
+	b.BundleByteThreshold = b.BundleCountThreshold * 200
+	b.BundleByteLimit = b.BundleCountThreshold * 1000
+	if o.TraceSpansBufferMaxBytes > 0 {
+		b.BufferedByteLimit = o.TraceSpansBufferMaxBytes
+	} else {
+		b.BufferedByteLimit = defaultBufferedByteLimit
+	}
+
+	e.bundler = b
 	e.uploadFn = e.uploadSpans
 	return e, nil
 }
 
 // ExportSpan exports a SpanData to Stackdriver Trace.
 func (e *traceExporter) ExportSpan(ctx context.Context, sd *export.SpanData) {
 	protoSpan := protoFromSpanData(sd, e.projectID)
-	e.uploadFn(ctx, []*tracepb.Span{protoSpan})
+	protoSize := proto.Size(protoSpan)
+	err := e.bundler.Add(&contextAndSpans{
+		ctx: ctx, 
+		spans: []*tracepb.Span{protoSpan},
+	}, protoSize)
+	switch err {
+	case nil:
+		return
+	case bundler.ErrOversizedItem:
```

Comment:
You need to use the `fallthrough` keyword here if you want to process both cases the same way.
Currently you are just ignoring this error.
See https://github.com/golang/go/wiki/Switch#fall-through